tree/30_generic_methods/service_restart_if.cf

```diff
@@ -31,14 +31,11 @@
 bundle agent service_restart_if(service_name, trigger_class)
 {
   vars:
-
     "canonified_service_name"      string => canonify("${service_name}");
-
     "old_class_prefix"             string => "service_restart_${canonified_service_name}";
-    "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
-    "class_prefix"      string => canonify(join("_", "promisers"));
     "args"                          slist => { "${service_name}",  "${trigger_class}" };
-
+    "report_param"                 string => join("_", args);
+    "class_prefix"                 string => canonify("service_restart_${report_param}");
```

Comment:
why isn't it service_restart_if ?

Reply:
because i took the original class_prefix,as it is easier to regenerate the new class prefix from the class prefix metadata and the arguments (for example if we want ot do it in the web interface)

Reply:
the one used in the web interface is old_class_prefix, not class_prefix